lib/route/link.c

```diff
@@ -358,6 +359,10 @@ static int link_clone(struct nl_object *_dst, struct nl_object *_src)
 		if ((err = rtnl_link_sriov_clone(dst, src)) < 0)
 			return err;
 
+	if (src->perm_addr)
+		if (!(dst->perm_addr = nl_data_clone(src->perm_addr)))
+			return -NLE_NOMEM;
```

Comment:
please squash the commits. This second commit just fixes a bug in the previous commit.